pennylane/optimize/rotosolve.py

```diff
@@ -132,340 +212,410 @@ class RotosolveOptimizer:
+                (``Sequence[float]``).
+                For each parameter, ``num_frequency`` take precedence over ``spectra``.
+            shifts (dict[dict]): Shift angles for the reconstruction per QNode parameter.
+                The keys have to be argument names of ``qnode`` and the inner dictionaries have to
+                be mappings from parameter indices to the respective shift angles to be used for
+                that parameter. For :math:`R` non-zero frequencies, there must be :math:`2R+1`
+                shifts given. Ignored if ``nums_frequency`` gives a number of frequencies
+                for the respective parameter in the QNode argument.
             full_output (bool): whether to return the intermediate minimized energy values from
-                the univariate optimization steps.
+                the univariate optimization substeps.
             **kwargs : variable length keyword arguments for the objective function.
 
         Returns:
-            tuple(list [array] or array, float): the new variable values :math:`x^{(t+1)}` and
-            the objective function output prior to the step.
+            list [array] or array: the new variable values :math:`x^{(t+1)}`.
             If a single arg is provided, list [array] is replaced by array.
-            list [float]: the intermediate energy values, only returned if ``full_output=True``.
+            float: the objective function output prior to the step.
+            list [float]: the intermediate objective values, only returned if
+            ``full_output=True``.
+
+        The optimization step consists of multiple substeps. For each substep,
+        one of the parameters in one of the QNode arguments is singled out, and the
+        objective function is considered as univariate function (i.e. function that
+        depends on a single scalar) of that parameter.
+        If ``nums_frequency`` states that there is only a single frequency, or ``spectra``
+        only contains one positive frequency, for a parameter, an analytic formula is
+        used to return the minimum of the univariate restriction.
+        For multiple frequencies, :func:`.fourier.reconstruct` is used to reconstruct
+        the univariate restriction and a numeric minimization is performed instead.
+        The latter minimization is performed using the ``optimizer`` passed to
+        ``RotosolveOptimizer`` at initialization.
+
+        .. note::
+
+            One of ``nums_frequency`` and ``spectra`` must contain information
+            about each parameter that is to be trained with ``RotosolveOptimizer``.
+            For each univariate reconstruction, the data in ``nums_frequency`` takes
+            precedence over the information in ``spectra``.
+
+        .. warning::
+
+            ``RotosolveOptimizer`` will only update parameters that are *explicitly*
+            markes as trainable via ``requires_grad``.
```

Comment:
```suggestion
            ``RotosolveOptimizer`` will only update parameters that are *explicitly*
            markes as trainable. Either via ``requires_grad`` if using Autograd or PyTorch,
            or by using `tf.Variable` tensors inside a `GradientTape` if using TensorFlow.
```

Reply:
(and JAX I presume doesn't work?)

Reply:
Actually... this optimizer is written for Autograd and torch only, currently, because it manually uses `getattr(arg, "requires_grad", True)` below. Maybe I can easily make this work for all interfaces (afaik the optimizer suite does not really have feature parity across interfaces yet, right?)

Reply:
all good for now! Let's open an issue though, so that users searching will find an open issue to add TF and JAX support here.

Reply:
> because it manually uses getattr(arg, "requires_grad", True)

wait, I thought `qml.math.requires_grad` was used? This is framework agnostic

Reply:
Oh dear, I looked at the _removed_ code :facepalm:

Reply:
Opened #2083 :)